cli/crates/backend/src/api/publish.rs

```diff
@@ -35,6 +38,9 @@ pub async fn publish(
         Some(PublishPayload::FederatedGraphCompositionError(FederatedGraphCompositionError { messages })) => {
             Ok(Err(messages.clone()))
         }
+        Some(PublishPayload::BranchDoesNotExistError(BranchDoesNotExistError { .. })) => {
+            Ok(Err(vec!["Provided branch does not exist in the project.".to_owned()]))
```

Comment:
I think the caller will print that with something like "Published with composition errors: - Provided branch does not exist....". If it's the case, we have to introduce an enum.

Reply:
Addressed. 🙂